vars/ip_whitelist_routes.json

```diff
@@ -0,0 +1,23 @@
+{
+  "preview": [
+    {"hostname": "dm-api-preview", "path": "/_metrics"},
+    {"hostname": "dm-search-api-preview", "path": "/_metrics"},
+    {"hostname": "dm-antivirus-api-preview", "path": "/_metrics"},
+    {"hostname": "dm-preview", "path": "/suppliers/_metrics"},
+    {"hostname": "dm-preview", "path": "/suppliers/opportunities/_metrics"}
```

Comment:
Is there a programmatic way to do this so that we don't have to update this later when we add `/_metrics` to the other FE apps?

We can get a list of hostnames and paths for an app from `cf curl /v2/apps/${APP_GUID}/routes | jq -r '.resources[].entity.host'` and `cf curl /v2/apps/${APP_GUID}/routes | jq -r '.resources[].entity.path'` (we use this method on the `map-route.sh` and `unmap-route.sh` scripts).

Reply:
While I don't like the idea of having to have another variables file just for this one script I also don't like the idea of hard coding `_metrics` in to everything. We could conceivably change `_metrics` to anything we like. Equally we could want to IP whitelist a route that is not `_metrics`.

It's a bit of a rock and a hard place solution. But I erred on the side of explicitly defining them.

Reply:
Is this info not already in the app manifest?

Reply:
No

1 There is no way of telling between routes we want protected from the manifest and routes we don't
2 The routes exist in the vars files as uninterpolated templates  eg `dm-{env}.apps.internal`

Reply:
We could add that information to the manifest? Rather than adding a new source of information. Then pass the expanded manifest to the script?

Reply:
I tried that, but they're for fundamentally different information.

We need to specify a subset of the routes that are defined in the manifest. _But_ we need to have them once they've been interpolated with the various variables.